mapswipe_workers/tests/integration/test_firebase_db_rules.py

```diff
@@ -0,0 +1,329 @@
+            [True, True],  # user username
+            [False, True],  # results public project
+            [False, False],  # results private project
+        ]
+
+        for i, endpoint in enumerate(self.endpoints):
+            path = endpoint[0].replace("<user_id>", user["localId"])
+            custom_arguments = endpoint[1]
+            read_access = test_get_endpoint(user, path, custom_arguments)
+            self.assertEqual(
+                read_access,
+                expected_access[i][0],
+                f"observed, expected, get {endpoint} {user['displayName']}",
+            )
+            write_access = test_set_endpoint(user, path)
+            self.assertEqual(
+                write_access,
+                expected_access[i][1],
+                f"observed, expected, set {endpoint} {user['displayName']}",
+            )
+
+    def test_access_as_team_member(self):
+        # sign in user with email and password to simulate app user
+        user = sign_in_with_email_and_password(
+            self.team_member.email, f"{self.team_member.display_name}_pw"
+        )
+
+        expected_access = [  # [read, write]
+            [True, False],  # public project query
+            [True, False],  # public project status attribute
+            [True, False],  # private project query
+            [True, False],  # private project status
+            [False, False],  # team
+            [True, False],  # teamName
+            [False, False],  # teamToken
+            [True, False],  # public group
+            [True, False],  # private group
+            [True, False],  # public task
+            [True, False],  # private task
+            [True, False],  # user
+            [True, False],  # user teamId
+            [True, True],  # user username
+            [False, True],  # results public project
+            [False, True],  # results private project
+        ]
+
+        for i, endpoint in enumerate(self.endpoints):
+            path = endpoint[0].replace("<user_id>", user["localId"])
+            custom_arguments = endpoint[1]
+            read_access = test_get_endpoint(user, path, custom_arguments)
+            self.assertEqual(
+                read_access,
+                expected_access[i][0],
+                f"observed, expected, get {endpoint} {user['displayName']}",
+            )
+            write_access = test_set_endpoint(user, path)
+            self.assertEqual(
+                write_access,
+                expected_access[i][1],
+                f"observed, expected, set {endpoint} {user['displayName']}",
+            )
+
+    def test_access_as_project_manager(self):
+        # sign in user with email and password to simulate app user
+        user = sign_in_with_email_and_password(
+            self.project_manager.email, f"{self.project_manager.display_name}_pw"
+        )
+
+        expected_access = [  # [read, write]
+            [True, False],  # public project query
```

Comment:
This is hairsplitting, and I don't think it really matters. I would have used a list of tuples here, because I think semantically it makes a bit more sense than a list of lists. But I wouldn't bother changing it now :)